stackslib/src/net/httpcore.rs

```diff
@@ -0,0 +1,1625 @@
+    pub fn try_parse_response(
+        &mut self,
+        request_handler_index: usize,
+        preamble: &HttpResponsePreamble,
+        body: &[u8],
+    ) -> Result<StacksHttpResponse, NetError> {
+        if preamble.status_code >= 400 {
+            return Self::try_parse_error_response(preamble, body);
+        }
+
+        let (_, _, parser) = self
+            .request_handlers
+            .get(request_handler_index)
+            .expect("FATAL: tried to use nonexistent response handler");
+        let payload = parser.try_parse_response(preamble, body)?;
+        let response = StacksHttpResponse::new(preamble.clone(), payload);
+        return Ok(response);
+    }
+
+    /// Handle an HTTP request by generating an HTTP response.
+    /// Returns Ok((preamble, contents)) on success.  Note that this could be an HTTP error
+    /// message.
+    /// Returns Err(..) on failure to decode or generate the response.
+    pub fn try_handle_request(
+        &mut self,
+        request: StacksHttpRequest,
+        node: &mut StacksNodeState,
+    ) -> Result<(HttpResponsePreamble, HttpResponseContents), NetError> {
+        let (decoded_path, _) = decode_request_path(&request.preamble().path_and_query_str)?;
+        let response_handler_index =
+            if let Some(i) = self.find_response_handler(&request.preamble().verb, &decoded_path) {
+                i
+            } else {
+                // method not found
+                return StacksHttpResponse::new_error(
+                    &request.preamble,
+                    &HttpNotFound::new(format!(
+                        "No such API endpoint '{} {}'",
+                        &request.preamble().verb,
+                        &decoded_path
+                    )),
+                )
+                .try_into_contents();
+            };
+
+        let (_, _, request_handler) = self
+            .request_handlers
+            .get_mut(response_handler_index)
+            .expect("FATAL: request points to a nonexistent handler");
+        let request_preamble = request.preamble.clone();
+        let request_result =
+            request_handler.try_handle_request(request.preamble, request.contents, node);
+        request_handler.restart();
+
+        let (response_preamble, response_contents) = match request_result {
+            Ok((rp, rc)) => (rp, rc),
+            Err(NetError::Http(e)) => {
+                return StacksHttpResponse::new_error(&request_preamble, &*e.into_http_error())
+                    .try_into_contents()
+            }
+            Err(e) => {
+                warn!("Irrecoverable error when handling request"; "path" => %request_preamble.path_and_query_str, "error" => %e);
+                return Err(e);
+            }
+        };
+        Ok((response_preamble, response_contents))
+    }
+
+    #[cfg(test)]
+    pub fn num_pending(&self) -> usize {
+        if self.reply.is_some() {
```

Comment:
If you wanted to remove the branches, you could change this to `u8::from(self.reply.is_some())` or `self.reply.is_some() as u8` since Rust now defines the memory representation for bool as a 1-byte value equal to either 0 or 1

Reply:
Fixed